### ISSUE TITLE
FPP 5.4.1 with Kulp OctoPlusCPU all over the place

### ISSUE THREAD

**Issue by nmiller0113** [FalconChristmas/fpp#1183]
# Please answer the following:
**FPP Version and Hardware**: 
FPP 5.4.1 with Kulp OctoPlus

**Describe the bug and Steps to reproduce**: 
I just re-imaged my Tune To sign with a new version of 5.4.1 and configured everything and uploaded the FSEQ for it to loop. When I was testing it I noticed that scrolling text was very jumpy and not smooth at all. I looked at the CPU and it was all over the place and often jumping up to 100%. I decided to delete the sequence and disable the cape and reboot...nothing changed...CPU is still all over the place. I used this same setup the last two years without issue and changed nothing but re-image to the latest version...just as I do every year. There are no plugins installed on this...it's a very very basic setup.

**Expected behavior**: 
CPU remains fairly steady and the text scrolling is smooth.

**Additional Attachments**
[FPP_Logs_20211203.1850.zip](https://github.com/FalconChristmas/fpp/files/7651775/FPP_Logs_20211203.1850.zip)

If applicable, add screenshots to help explain your problem
<img width="1891" alt="Screen Shot 2021-12-03 at 10 41 58 AM" src="https://user-images.githubusercontent.com/1048868/144657750-bdf05ec1-d339-4f02-905b-509156951264.png">

**Comment by darylc** [comment]
Please send us the information from help->troubleshooting->cpu utilization

**Comment by nmiller0113** [comment]
Here you go @darylc 

CPU Utilization:    top -bn1 | head -20
top - 16:46:36 up 39 min,  0 users,  load average: 1.84, 0.59, 0.39
Tasks: 114 total,   2 running, 112 sleeping,   0 stopped,   0 zombie
%Cpu(s): 31.6 us, 68.4 sy,  0.0 ni,  0.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :    480.3 total,    227.3 free,     97.0 used,    156.0 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.    321.3 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
   68 root     -51   0       0      0      0 S   9.8   0.0   0:20.55 irq/48-48+
 3858 fpp       20   0    5160   3004   2668 R   6.6   0.6   0:00.12 ntpq
 3874 root      20   0    4656   2564   2204 S   6.6   0.5   0:00.04 sudo
 3876 root      20   0    4656   2580   2220 S   6.6   0.5   0:00.04 sudo
  134 root      20   0   35292   5656   4740 S   4.9   1.2   0:02.04 systemd-j+
 3863 fpp       20   0    4664   2132   1736 R   4.9   0.4   0:00.10 top
 3070 fpp       20   0  209200  17948  12752 S   3.3   3.6   0:00.54 apache2
 3512 fpp       20   0  199932  16612  11644 S   3.3   3.4   0:00.24 apache2
   10 root      20   0       0      0      0 I   1.6   0.0   0:02.07 rcu_preem+
  166 message+  20   0    5600   2948   2472 S   1.6   0.6   0:01.31 dbus-daem+
  181 root      20   0   23576   3000   2252 S   1.6   0.6   0:00.43 rsyslogd
  285 root      20   0   24604   7996   6868 S   1.6   1.6   0:15.37 fppoled
  622 ntp       20   0    7172   2892   2420 S   1.6   0.6   0:01.04 ntpd

**Comment by nmiller0113** [comment]
Any thoughts from you or @dkulp? My show started tonight and my P5 Tune To sign looks pretty bad :( Literally used this sign the last 3 years...changed nothing but the re-image.

**Comment by darylc** [comment]
I'd suggest troubleshooting this in the xlights zoom room, I can't see anything obvious from the data you've provided.

**Comment by nmiller0113** [comment]
@darylc I don’t think anyone in the zoom room would be able to assist with this. It’s a purely FPP related issue. My CPU is jumping regularly to 100%…multiple times a minute. The us (user) and sy (system) reflect it in my capture. How is that regular/not obvious? This is not xLights related. This is a re-imaged build that I’m running, which I’ve done 100’s of times (since v1) on many instances of FPP I’m running simultaneously…this is the only one out of 5 instances currently running FPP 5.4.1 behaving this way. This is the only one with an OctoPlus. The cape is disabled currently and no sequence is running on it. It’s not in bridged mode…it’s doing absolutely nothing and CPU is hitting 100%. I’m sorry, I’m not trying to be a pain, but I don’t feel this will be solved in the zoom room…this is not a configuration issue. Something is causing this and I’ve been monitoring the OS services and unable to pinpoint it.

I’m happy to jump on a zoom with someone who could dig into the OS on this, but the xLights zoom room is not the place for that. Happy to provide more configs or data. I should mention that I just re-imaged it again, full wipe, and it’s still doing this before I’ve done so much as expanded the storage.

**Comment by cpinkham** [comment]
Can you try connecting via SSH and running 'top'?  When CPU jumps to 100, what is using the CPU?  I see a couple 'sudo' processes in your process listing above.  I'm curious what those are since there shouldn't be a bunch of sudo commands running if the system is just sitting there idle.

**Comment by nmiller0113** [comment]
Chris, I was actually digging in on that earlier and ran top and I didn’t see any root cause, though I’m happy to ssh in and see if I can capture anything. I did see the sudo processes as well and thought it was interesting, but knowing nothing about how it should look in FPP I couldn’t determine if it was a false positive suspect or not.

Happy to jump on a zoom as well and share my terminal to do what you feel would be valuable with you and/or @dkulp

**Comment by cpinkham** [comment]
Try running 'top -c' to see the full command line to see what those sudo processes are if they are still showing up.  I will have to defer to Dan if the irq line is still at the top, but if you see something else then I may be able to help.  Do you have any plugins installed on this system?

**Comment by nmiller0113** [comment]
@cpinkham it's hard to see as there are moments where the CPU id goes to zero, or drops very close to it, but it's only for a second so it's hard to catch what it is...but this would align with my stuttering in scrolling text.

No plugins installed on this system...this build actually has nothing running on it at all. No configuration currently (fresh re-image) and it's still pegging its CPU from time to time. I will say that in monitoring top -c, every time I do the troubleshooting steps @darylc mentioned "Please send us the information from help->troubleshooting->cpu utilization" my CPU id drops to 0. This could all be normal, but not sure. All I know is I never had stuttering in scrolling test like this ever in the past...so I'm not sure why this is happening now with the exact same setup as last year.

**Comment by darylc** [comment]
@nmiller0113 You seem in a hurry to fix this thats why I suggested the zoom room. Diagnosing things like this via github will take weeks.  There are multiple people in there who do delve into the OS if they are around at the time.  Since you are the only person we've heard of having this issue it must somehow be related to your environment.  Like the Captain I am also suspicious of the sudo's but we're working largely blind here.  Have you tried a different physical SD card?

**Comment by nmiller0113** [comment]
@darylc my apologies if I did come across the wrong way...I've got quite a bit of linux and system side experience and assumed that I was past OS troubleshooting and into the realm of FPP OS level troubleshooting...especially considering I even tried a fresh install with zero configuration changes. You do bring up a good point though regarding the SD card. That is something else I changed this year...I upgraded them all to 32GB, but these cards are also capable of higher speeds. Perhaps that's the issue considering all of my other FPP instances are Pi 4's...this is the only BBB. I can test that theory tomorrow. Thank you again!

**Comment by dkulp** [comment]
You will definitely need to be ssh'd in and running top or similar to see what processes are using the CPU.    I was monitoring my P5 tuneto sign last night (on a PocketScroller though) an it stayed very steady at about 30% while the show was a running.  When the show stopped, dropped right to about 1.5% (most of that was the top process itself).

**Comment by nmiller0113** [comment]
@darylc @cpinkham @dkulp, here’s the latest. Last night was the opening of my show, and I didn’t have an opportunity to swap the SD card. Funny thing though, it stopped stuttering after a few mins…all by itself. Any ideas? It happens every time I start the FSEQ effect fresh…and then the stuttering stops.

I tested it today and put the original SD card back in it re-imaged with 5.4.1. No difference, so it’s not the card.

Regarding the CPU jumping. As it’s turns out, the CPU jumps up to 100% or close to it every time I hit the WebUI on the upgrade page where it shows the CPU, though the troubleshooting page also does it. It seems that moving around the WebUI in general seems to hit the CPU pretty hard on the OctoPlus BBB. Versus my Pi4 instances that rarely hit 10% when I view those pages…which is why I originally thought that was the issue. My apologies. Once I monitored top from a shell, it stayed pretty flat, unless I was moving around the WebUI.

Things seem to be working well now, just stuttering in the beginning. I’m assuming that’s not expects, but it’s not the end of the world. So I’ll keep this open and see what you think we should do. Happy to deep dive this following the holiday season when folks have more time on their hands :)
